src/createAnimatedComponent.tsx

```diff
@@ -533,43 +535,35 @@ export default function createAnimatedComponent(
       setLocalRef: (ref) => {
         // TODO update config
         const tag = findNodeHandle(ref);
-        if (
-          (this.props.layout || this.props.entering || this.props.exiting) &&
-          tag != null
-        ) {
+        const { layout, entering, exiting } = this.props;
+        if ((layout || entering || exiting) && tag != null) {
```

Comment:
Is this rewrite potentially the right place to also fix #3662? Specifically, `entering` and `exiting` animations are only built/configured when setting the ref; they aren't being updated by `componentDidUpdate` when their values change.

Thanks for attempting this fix, @kmagiera & @piaskowyk .